lms/djangoapps/homepage/views.py

```diff
@@ -1,5 +1,28 @@
+import branding
+from common.lib.nodebb_client.client import NodeBBClient
+from custom_settings.models import CustomSettings
 from edxmako.shortcuts import render_to_response
 
 
 def home_page(request):
-    return render_to_response("homepage/homepage.html", {})
+    featured_courses = get_featured_courses()
+    status_code, featured_categories = NodeBBClient().categories.featured()
+
+    context = {
+        'courses': featured_courses,
+        'categories': featured_categories if status_code not in range(400, 499) else [],
```

Comment:
This `if` statement seems unnecessary and might not be sufficient in some cases. For instance if we get 500 status code. what about using something like this?
```
'categories': featured_categories or []
```

Reply:
Do not work in a scenario when we get **400**, In that case we get **Not Found** in featured categories